test/federated/test_train_config.py

```diff
@@ -329,3 +355,165 @@ def test_send_model_and_loss_fn(workers):
     assert alice.train_config._loss_fn_id == loss_fn_id
 
     sy.ID_PROVIDER.pop = orig_func
+
+
+@pytest.mark.skipif(
+    torch.__version__ >= "1.1",
+    reason="bug in pytorch version 1.1.0, jit.trace returns raw C function",
+)
+@pytest.mark.asyncio
+async def test_train_config_with_jit_trace_async(hook, start_proc):  # pragma: no cover
+    kwargs = {"id": "async_fit", "host": "localhost", "port": 8777, "hook": hook}
+    # data = torch.tensor([[0.0, 1.0], [1.0, 0.0], [1.0, 1.0], [0.0, 0.0]], requires_grad=True)
+    # target = torch.tensor([[1.0], [1.0], [0.0], [0.0]], requires_grad=False)
+    # dataset_key = "xor"
+    data, target = utils.create_gaussian_mixture_toy_data(100)
+    dataset_key = "gaussian_mixture"
+
+    mock_data = torch.zeros(1, 2)
+
+    # TODO check reason for error (RuntimeError: This event loop is already running) when starting websocket server from pytest-asyncio environment
```

Comment:
We had similar issues in jupyter since jupyter uses asyncio under the hood. This might be the same issue